homeassistant/components/octoprint/sensor.py

```diff
@@ -102,6 +124,18 @@ def __init__(
         self._icon = icon
         _LOGGER.debug("Created OctoPrint sensor %r", self)
 
+    @property
+    def device_info(self):
+        """Device info."""
+        return {
+            "identifiers": {(COMPONENT_DOMAIN, self.api.api_url)},
+        }
+
+    @property
+    def unique_id(self):
+        """Return a unique id."""
+        return f"{self._name}-{self.api.api_url}"
```

Comment:
This doesn't look like it will be long term stable

Is there a better option for the unique id here?
https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements

It may be better to use the config entry id here instead

Reply:
We should use the unique ID provided by Octoprint itself.

Reply:
Yeah, there is a better UUID supplied by the tracking plugin of Octoprint. It isn't on the existing calls the API was making so for the purposes of this PR I was trying to keep it a bit simpler. Was planning on swapping it over next. The size of this PR is getting a bit bigger than I like personally given this is not targeting develop. I can add the change here into this PR as well however if folks feel is it warrented.

Reply:
I think this emphasizes why it might have been better to first put in a library to abstract the API, as right now, it would put patterns in, we generally don't want to have.

Reply:
I will push the library changes up tonight into this PR. Should include everything but the auth changes.

Reply:
I really think you should not. I think it is better to make a separate PR just first abstract the API logic away into a lib. Next, rebase this PR onto it. The API logic to replace the current logic in the existing integration is really minimal.

Reply:
PR: https://github.com/home-assistant/core/pull/46611
Let me know if I took the wrong approach with it, still getting used to working with python.

I can rebase this one once that one gets some eyes on it

Reply:
Swapped over to the the UUID